cc/main.cc

```diff
@@ -142,55 +151,83 @@ void WriteSgf(const std::string& output_dir, const std::string& output_name,
   MG_CHECK(file::SetContents(output_path, sgf_str));
 }
 
-}  // namespace
+void ParseMctsPlayerOptionsFromFlags(MctsPlayer::Options* options) {
+  options->inject_noise = FLAGS_inject_noise;
+  options->soft_pick = FLAGS_soft_pick;
+  options->random_symmetry = FLAGS_random_symmetry;
+  options->resign_threshold = FLAGS_resign_threshold;
+  options->batch_size = FLAGS_batch_size;
+  options->komi = FLAGS_komi;
+  options->random_seed = FLAGS_seed;
+}
 
 void SelfPlay() {
-  Random rnd;
-
-  // Initialize the network.
-  DualNet dual_net;
-  dual_net.Initialize(FLAGS_model);
+  auto dual_net = absl::make_unique<DualNet>();
+  dual_net->Initialize(FLAGS_model);
 
-  // Initialize the MCTS player.
   MctsPlayer::Options options;
-  options.random_seed = FLAGS_seed;
-  options.komi = FLAGS_komi;
-  options.inject_noise = FLAGS_inject_noise;
-  options.soft_pick = FLAGS_soft_pick;
-  options.random_symmetry = FLAGS_random_symmetry;
-  options.batch_size = FLAGS_batch_size;
-  if (rnd() < FLAGS_holdout_pct) {
+  ParseMctsPlayerOptionsFromFlags(&options);
+  Random rnd;
```

Comment:
should rnd be seeded with option.random_seed?

Reply:
I don't think so. random_seed controls the determinism of the moves played, not whether the game is selected for holdout etc. Updated the --seed flag's help text.